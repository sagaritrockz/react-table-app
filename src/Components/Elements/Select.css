.select {
    display: inline-block;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
    text-transform: capitalize;
}
.select-custom {
    padding: 5px 10px 8px;
    margin: 0 5px;
}
