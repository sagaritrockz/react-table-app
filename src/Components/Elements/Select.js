import React from "react"
import "./Select.css"

const Select = ({ list, change, value, name, show }) => {

    const onChange = e => {
        e.preventDefault();
        change(e.target.name, e.target.value);
    };
    
    if (show) {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                <option value={0}>Select</option>
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    } else {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    }
};

Select.defaultProps = {
    list: [],
    change: f => f, 
    value: '', 
    name: '',
    show: false
}

export default Select;
