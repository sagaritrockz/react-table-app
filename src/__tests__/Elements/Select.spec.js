import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Select from '../../Components/Elements/Select'

Enzyme.configure({adapter: new Adaptar()});

describe('Select', () => {
    it('should render correctly with empty', () => {
        const output = shallow(
            <Select
                list={[]}
                value={"0"}
                change={function(){}}
                name="MockName"
                defaultOption="Select"
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
  
    it('should render correctly for if condition', () => {
        const list = ['Region', 'IncomeLevels', 'LendingTypes'];
        const output = shallow(
            <Select
                list={list}
                value={"0"}
                change={function(){}}
                name="MockName"
                defaultOption="Select"
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should render correctly for else condition', () => {
        const list = ['Region', 'IncomeLevels', 'LendingTypes'];
        const output = shallow(
            <Select
                list={list}
                value={"0"}
                change={function(){}}
                name="MockName"
                defaultOption=""
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});