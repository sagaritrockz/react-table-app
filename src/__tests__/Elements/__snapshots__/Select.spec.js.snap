// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly for else condition 1`] = `
<select
  className="select select-custom"
  name="MockName"
  onChange={[Function]}
  value="0"
>
  <option
    key="0"
    value={1}
  >
    Region
  </option>
  <option
    key="1"
    value={2}
  >
    IncomeLevels
  </option>
  <option
    key="2"
    value={3}
  >
    LendingTypes
  </option>
</select>
`;

exports[`Select should render correctly for if condition 1`] = `
<select
  className="select select-custom"
  name="MockName"
  onChange={[Function]}
  value="0"
>
  <option
    value={0}
  >
    Select
  </option>
  <option
    key="0"
    value={1}
  >
    Region
  </option>
  <option
    key="1"
    value={2}
  >
    IncomeLevels
  </option>
  <option
    key="2"
    value={3}
  >
    LendingTypes
  </option>
</select>
`;

exports[`Select should render correctly with empty 1`] = `
<select
  className="select select-custom"
  name="MockName"
  onChange={[Function]}
  value="0"
>
  <option
    value={0}
  >
    Select
  </option>
</select>
`;
